### ISSUE TITLE
5.3.0 fail npm i on mac os x

### ISSUE THREAD

**Issue by nbeny** [justadudewhohacks/opencv4nodejs#633]
npm i opencv4nodejs

ERR! RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stderr maxBuffer length exceeded

this error come on the version opcv4nodejs@5.3.0

I use Mac OS X (Majove Version 10.14.3)

the version 5.2.0 working fine

**Comment by sirShakir** [comment]
I am having same issue but installing on a RPI.

**Comment by nbeny** [comment]
juste do
```
npm i opensv4nodejs@5.2.0
```

**Comment by sirShakir** [comment]
Thanks man. What could have happened between 5.2 and 5.3 that crashes the memory with this build?

**Comment by nbeny** [comment]
I do not really know, the error handling buffer is exceeded. Like we see a lot of warning at the compilation and he can't print everything in the buffer.

**Comment by snakelecaps** [comment]
hey guys, do you have the error `Error: Cannot find module /node_modules/opencv4nodejs/build/Release/opencv4nodejs'`  ? I dont know why i got it.... Getting crazy

**Comment by GUIEEN** [comment]
Here is my settings for installing this `opencv4nodejs@5.3.0` library

```
$ nvm ls
        v9.11.2
       v10.16.3
->      v12.9.0

$ export OPENCV4NODEJS_DISABLE_AUTOBUILD=1
$ npm i opencv4nodejs                     
```
I've used almost 2h just for installing this one but the key was 
`export OPENCV4NODEJS_DISABLE_AUTOBUILD=1`
before installing nor the npm version, node version etc.

Hope this could help someone struggling like me ;)

**Comment by SamuIos** [comment]
Any idea? i'm stuck too :(
Mac mojave version 10.14.6

> tr_emplace' requested here
    ::new(__hold2.get()) _CntrlBlk(__a2, _VSTD::forward<_Args>(__args)...);
                         ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/memory:4710:29: note: in instantiation of function template specialization 'std::__1::shared_ptr<CoreBindings::Solve>::make_shared<>' requested here
    return shared_ptr<_Tp>::make_shared(_VSTD::forward<_Args>(__args)...);
                            ^
/usr/local/Cellar/node/12.10.0/lib/node_modules/opencv4nodejs/node_modules/native-node-utils/src/ObjectWrap.h:59:23: note: in instantiation of function template specialization 'std::__1::make_shared<CoreBindings::Solve>' requested here
                        auto worker = std::make_shared<WorkerImpl>();
                                           ^
../cc/core/Mat.cc:758:7: note: in instantiation of function template specialization 'FF::ObjectWrapTemplate<Mat, cv::Mat>::syncBinding<CoreBindings::Solve>' requested here
        Mat::syncBinding<CoreBindings::Solve>("Solve", info);
             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/memory:3710:23: note: qualify call to silence this warning
    __data_.second().~_Tp();
                      ^
2 warnings generated.
In file included from ../cc/core/Rect.cc:1:
In file included from ../cc/core/Rect.h:1:
In file included from /usr/local/include/opencv4/opencv2/core.hpp:52:
In file included from /usr/local/include/opencv4/opencv2/core/cvdef.h:695:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/array:112:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/algorithm:644:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/memory:3710:5: warning: ERR! RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: **stderr maxBuffer length exceeded** 
>

**Comment by muonkiepdoiemhoai** [comment]
làm sao để khắc phục vấn đề

**Comment by neolp** [comment]
Same problem with RPi 3, Linux 4.19.75-v7+ #1270 SMP Tue Sep 24 18:45:11 BST 2019 armv7l
npm -v 6.9.0
node -v v10.16.3
cmake version 3.13.4

**Comment by sirShakir** [comment]
But now the application is crashing often from segmentation fualt

**Comment by zaquest** [comment]
Same issue arch linux, npm 6.12.0, node 12.13.0, cmake 3.15.4. And I can't use 5.2.0 anymore, since arch updated libImath :(

**Comment by justadudewhohacks** [comment]
Probably have to increase the maxBuffer option for the gyp processes stdout buffer, since the install script spawns a gyp process from node rather then directly from npm since 5.3.0:
https://github.com/justadudewhohacks/opencv4nodejs/blob/master/install/install.js#L83-L86

Not sure why this error occurs for you but does not for any of the CI tasks, probably due to your environment a lot of warnings are generated when compiling the module. Anyways I think [this](https://stackoverflow.com/questions/23429499/stdout-buffer-issue-using-node-child-process) solution should fix it. Will get a fix pushed this weekend.

**Comment by zaquest** [comment]
Failed to find that exec call on my own, thank you! Increasing maxBuffer to 1024^3 fixed the issue for me. It is indeed likely that compilation on arch produces more warnings than usually since they often update gcc, which in turn often rolls out new warnings. It is hard to say what the appropriate maxBuffer value could be - it's different for every environment. For me stderr and stdout lengths where 3070605 and 7635 respectively.
The current workaround for me, that doesn't require modifying the source code, is `CFLAGS="-w" CXXFLAGS="-w" npm install opencv4nodejs`. This forces cmake to pass `-w` to gcc, which suppresses most of the warnings.

**Comment by justadudewhohacks** [comment]
This issue is fixed with 5.5.0. Please let me know, if you are still facing this error for some reason.
